db/migrate/20200526122730_populate_completed_section_booleans.rb

```diff
@@ -0,0 +1,50 @@
+class PopulateCompletedSectionBooleans < ActiveRecord::Migration[6.0]
+  def up
+    submitted_applications = ApplicationForm.where.not(submitted_at: nil)
+
+    submitted_applications.includes(application_choices: %i[course_option course]).each do |application_form|
+      fields_to_update = {
+        personal_details_completed: true,
+        contact_details_completed: true,
+        english_gcse_completed: true,
+        maths_gcse_completed: true,
+        training_with_a_disability_completed: true,
+        safeguarding_issues_completed: true,
+        becoming_a_teacher_completed: true,
+        subject_knowledge_completed: true,
+        interview_preferences_completed: true,
+        references_completed: true,
+      }
+
+      if application_form.science_gcse_needed?
+        fields_to_update.merge!({
+          science_gcse_completed: true,
+        })
+      end
+
+      application_form.update!(fields_to_update)
+    end
+
+    in_progress_application_forms = ApplicationChoice.includes(:application_form).unsubmitted.map(&:application_form).uniq
+    started_application_forms = in_progress_application_forms.reject(&:blank_application?)
+
+    started_application_forms.each do |application_form|
```

Comment:
~~Doesn't really matter with data volumes as they are but this should be a `find_each` to batch the results.~~

Actually scratch that it's a bit more complicated than that because `find_each` would be needed before the `map` and `reject`.